floor/test/integration/stream_query_test.dart

```diff
@@ -137,16 +137,20 @@ void main() {
           actual,
           emitsInOrder(<List<Dog>>[
             [], // initial state,
+            [], // after inserting person1,
+            [], // after inserting person2,
             [dog1], // after inserting dog1
             [dog1], // after inserting dog2
-            //[], // after removing person1. Does not work because
-            // ForeignKey-relations are not considered yet (#321)
+            [], // after removing person1.
```

Comment:
❤️ Good job! This is great!

Reply:
Here I found out that this is still suboptimal, since it will be also triggered if a person is inserted, even if that will never change `dog`. :laughing: 

This is what I meant in #373 .